pkg/runtime/unstructured/converter.go

```diff
@@ -232,3 +232,176 @@ func (c *Converter) structFromUnstructured(sv, dv reflect.Value) error {
 func (c *Converter) interfaceFromUnstructured(sv, dv reflect.Value) error {
 	return fmt.Errorf("Interface conversion unsupported")
 }
+
+var (
+	marshalerType          = reflect.TypeOf(new(encodingjson.Marshaler)).Elem()
+	mapStringInterfaceType = reflect.TypeOf(map[string]interface{}{})
+)
+
+func (c *Converter) ToUnstructured(obj runtime.Object, u *map[string]interface{}) error {
```

Comment:
I'm not sure what value having the `Converter` type brings.  I'd recommend all the private methods just become private functions, which dramatically simplifies this unless there's something usefel you'll be adding to Converter.  All of the leaf methods that don't have a reason should just be functions.